### PR TITLE
Improving the login layout when IE is in intranet mode

### PR DESCRIPTION
**Pull request by rhamilto**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1298750

### CODE REVIEW

**Comment by jwforres** [comment]
@rhamilto if media queries are not supported that tells me even more of the console is broken than we realized, if you shrink things down to small desktop is it a mess?

**Comment by jwforres** [comment]
cc @spadgett

**Comment by spadgett** [comment]
@rhamilto Does adding 

```html
 <meta http-equiv="X-UA-Compatible" content="IE=edge"> 
```

to the document head help?

**Comment by jwforres** [comment]
ahhhh right that is how we forced the document mode

**Comment by rhamilto** [comment]
@spadgett, yep!  That's the missing key.  The login page does not have the meta element, but the rest of the console does.  Will update the PR.

Is there a way for me to test the PR itself?  I've been working off local files...

**Comment by spadgett** [comment]
You can try

```
$ OS_BUILD_ENV_PRESERVE="_output/local" hack/env OS_ONLY_BUILD_PLATFORMS="linux/amd64" WHAT=cmd/openshift make
$ hack/build-local-images.py origin
$ oc cluster up --version=latest
```

**Comment by jwforres** [comment]
@rhamilto so we actually have several other templates for the oauth server, and I guarantee they probably are ALL missing this line

**Comment by jwforres** [comment]
@liggitt FYI since this will be touching all of the oauth server's HTML pages (will wait for 3.7)

File: pkg/auth/server/login/templates.go [hunk 1]
```diff
@@ -83,6 +83,7 @@ const defaultLoginTemplateString = `<!DOCTYPE html>
   <head>
     <title>Login - OpenShift Origin</title>
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
```
Comment (spadgett) [inline pkg/auth/server/login/templates.go]: This should be the first meta tag. Best to put it immediately after head.

**Comment by jwforres** [comment]
@rhamilto so @liggitt also has an oauth access granting page which isn't based off this same theme, but could still have issues, dont remember what the name of that file is offhand, maybe @deads2k knows?

**Comment by rhamilto** [comment]
> @rhamilto so @liggitt also has an oauth access granting page which isn't based off this same theme, but could still have issues, dont remember what the name of that file is offhand

I think it's `pkg/auth/server/grant/templates.go`

I've searched the repo for the following and updated accordingly:

contain "data:image"

* pkg/auth/server/errorpage/templates.go [UPDATED]
* pkg/auth/server/login/templates.go [UPDATED]
* pkg/auth/server/selectprovider/templates.go [UPDATED]

contain "@media"

* cmd/cluster-capacity/go/src/github.com/kubernetes-incubator/cluster-capacity/* (documentation pages)
* examples/atomic-registry/allinone/registry-login-template.html [UPDATED]
* examples/atomic-registry/systemd/templates/registry-login-template.html [UPDATED]
* pkg/auth/server/errorpage/templates.go [UPDATED]
* pkg/auth/server/grant/templates.go [UPDATED]
* pkg/auth/server/login/templates.go [UPDATED]
* pkg/auth/server/selectprovider/templates.go [UPDATED]
* pkg/auth/server/tokenrequest/endpoints.go (not a full html file and only contains one rule--.nowrap)

I think we should be good.

File: pkg/auth/server/grant/templates.go [hunk 2]
```diff
@@ -9,8 +9,9 @@ const defaultGrantTemplateString = `<!DOCTYPE html>
 <html>
 
 <head>
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
     <title>
-      Authorize 
+      Authorize
```
Comment (jwforres) [inline pkg/auth/server/grant/templates.go]: probably want to be careful about these whitespace changes, it might be affecting the final output

**Comment by jwforres** [comment]
[test]

**Comment by jwforres** [comment]
I think we might get a clean merge of this change with the existing OCP template overrides, so we may not need a separate PR for OCP

**Comment by liggitt** [comment]
LGTM

**Comment by smarterclayton** [comment]
Low risk, big win [severity:blocker] [merge]

**Pull request merged**